deepchecks/vision/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,208 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module of weak segments performance check."""
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+import torch
+
+from deepchecks import ConditionCategory, ConditionResult
+from deepchecks.core import CheckResult, DatasetKind
+from deepchecks.core.check_result import DisplayMap
+from deepchecks.core.errors import DeepchecksNotSupportedError, DeepchecksProcessError, DeepchecksValueError
+from deepchecks.tabular import Dataset
+from deepchecks.tabular.context import _DummyModel
+from deepchecks.utils.performance.weak_segment_abstract import WeakSegmentAbstract
+from deepchecks.utils.single_sample_metrics import per_sample_cross_entropy
+from deepchecks.utils.strings import format_number, format_percent
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.iou_utils import per_sample_mean_iou
+from deepchecks.vision.metrics_utils.semantic_segmentation_metrics import per_sample_dice
+from deepchecks.vision.task_type import TaskType
+from deepchecks.vision.utils.image_properties import default_image_properties
+from deepchecks.vision.utils.vision_properties import PropertiesInputType
+
+
+class WeakSegmentsPerformance(SingleDatasetCheck, WeakSegmentAbstract):
+    """Search for segments with low performance scores.
+
+    The check is designed to help you easily identify weak spots of your model and provide a deepdive analysis into
+    its performance on different segments of your data. Specifically, it is designed to help you identify the model
+    weakest segments in the data distribution for further improvement and visibility purposes.
+
+    In order to achieve this, the check trains several simple tree based models which try to predict the error of the
+    user provided model on the dataset. The relevant segments are detected by analyzing the different
+    leafs of the trained trees.
+
+    Parameters
+    ----------
+    scorer: Optional[Callable], default: None
```

Comment:
Should clarify that here the user must provide a scorer that return per sample score
Would also change it to a dict of 'scorer_name': Callable